File: app/src/main/java/com/wegielek/signalychinese/database/FlashCards.java
package com.wegielek.signalychinese.database;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "flash_cards")
public class FlashCards {
    @PrimaryKey(autoGenerate = true)
    public int uid;

    @NonNull
    @ColumnInfo(name = "group")
    public String group = "";

    @NonNull
    @ColumnInfo(name = "traditional_sign")
    public String traditionalSign = "";

    @NonNull
    @ColumnInfo(name = "simplified_sign")
    public String simplifiedSign = "";

    @NonNull
    @ColumnInfo(name = "pronunciation")
    public String pronunciation = "";

    @NonNull
    @ColumnInfo(name = "pronunciation_phonetic")
    public String pronunciationPhonetic = "";

    @NonNull
    @ColumnInfo(name = "translation")
    public String translation = "";
}

File: app/src/main/java/com/wegielek/signalychinese/database/FlashCards.kt
package com.wegielek.signalychinese.database

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "flash_cards")
class FlashCards {
    @PrimaryKey(autoGenerate = true)
    var uid = 0

    @ColumnInfo(name = "group")
    var group = ""

    @ColumnInfo(name = "traditional_sign")
    var traditionalSign = ""

    @ColumnInfo(name = "simplified_sign")
    var simplifiedSign = ""

    @ColumnInfo(name = "pronunciation")
    var pronunciation = ""

    @ColumnInfo(name = "pronunciation_phonetic")
    var pronunciationPhonetic = ""

    @ColumnInfo(name = "translation")
    var translation = ""
}
